code/game/objects/items/storage/boxes.dm

```diff
@@ -1357,3 +1357,141 @@
 	desc = "A bag containing fresh, dry coffee robusta beans. Ethically sourced and packaged by Waffle Corp."
 	beantype = /obj/item/reagent_containers/food/snacks/grown/coffee/robusta
 
+/obj/item/storage/box/ice_cream_carton
+	icon_state = "ice_cream"
+	icon = 'icons/obj/food/containers.dmi'
+	name = "Big Top plain ice cream carton"
+	desc = "A classic ice cream brand; this carton contains plain ice cream."
+
+	//What flavor will be inside the carton
+	var/ice_cream_flavor = /obj/item/reagent_containers/food/snacks/ice_cream_scoop
+
+/obj/item/storage/box/ice_cream_carton/Initialize(mapload)
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_items = 7
+	STR.set_holdable(list(/obj/item/reagent_containers/food/snacks/ice_cream_scoop))
+
+/obj/item/storage/box/ice_cream_carton/PopulateContents()
+	for(var/i in 1 to 7)
+		new ice_cream_flavor(src)
+
+/obj/item/storage/box/ice_cream_carton/attackby(obj/item/A, mob/user, params)
+	//Allow for name and desc to be changed with pen
+	if(istype(A, /obj/item/pen))
+		var/choice = input(usr, "Choose which flavor to change to", "Changing Carton Flavor") as null|anything in list("Plain", "Vanilla", "Chocolate", "Strawberry", "Blue", "Lemon Sorbet", "Caramel", "Banana", "Orange Creamsicle", "Peach", "Cherry Chocolate", "Meat Lover's")
+		switch(choice)
+			if("Plain")
+				name = "Big Top plain ice cream carton"
+				desc = "A classic ice cream brand; this carton contains plain ice cream."
```

Comment:
refactor this into using datums so it's easier to expand upon
or if you don't know how to do that
at the very least create defines for the names

Reply:
Don't know how to do it with datums so I did it with defines.